src/tapis-app/Files/_components/Toolbar/UploadModal/UploadModal.tsx

```diff
@@ -0,0 +1,241 @@
+import { useEffect, useState, useCallback, useReducer } from 'react';
+import { Button } from 'reactstrap';
+import { GenericModal, LoadingSpinner, Icon } from 'tapis-ui/_common';
+import { SubmitWrapper } from 'tapis-ui/_wrappers';
+import { ToolbarModalProps } from '../Toolbar';
+import { useUpload } from 'tapis-hooks/files';
+import { focusManager } from 'react-query';
+import { useDropzone } from 'react-dropzone';
+import styles from './UploadModal.module.scss';
+import { FileListingTable } from 'tapis-ui/components/files/FileListing';
+import { Files } from '@tapis/tapis-typescript';
+import { Column } from 'react-table';
+import sizeFormat from 'utils/sizeFormat';
+import { useMutations } from 'tapis-hooks/utils';
+import { InsertHookParams } from 'tapis-hooks/files/useUpload';
+import Progress from 'tapis-ui/_common/Progress';
+
+export enum FileOpEventStatus {
+  loading = 'loading',
+  progress = 'progress',
+  error = 'error',
+  success = 'success',
+  none = 'none',
+}
+
+export type FileOpState = {
+  [key: string]: {
+    status: FileOpEventStatus;
+    progress?: number;
+  };
+};
+
+type UploadModalProps = ToolbarModalProps & {
+  maxFileSizeBytes?: number;
+};
+
+const UploadModal: React.FC<UploadModalProps> = ({
+  toggle,
+  path,
+  systemId,
+  maxFileSizeBytes = 524288000,
+}) => {
+  const [files, setFiles] = useState<Array<File>>([]);
+
+  const onProgress = (uploadProgress: number, file: File) => {
+    dispatch({
+      key: file.name,
+      status: FileOpEventStatus.progress,
+      progress: uploadProgress,
+    });
+  };
+
+  const isValidFile = useCallback(
```

Comment:
This shouldn't be a callback. If the maxFileSizeBytes param changes the entire component will get remounted and the a new function would be generated with the new value anyway.

Reply:
I was forced to put it in a callback by the linter for some reason. I'll remove it to see what happens now.